guides/plugins/plugins/framework/data-handling/add-data-indexer.md

```diff
@@ -0,0 +1,263 @@
+* `ProductIndexer`
+* `ProductStreamIndexer`
+* `PromotionIndexer`
+* `RuleIndexer`
+* `MediaIndexer`
+* `MediaFolderIndexer`
+* `MediaFolderConfigurationIndexer`
+* `SalesChannelIndexer`
+* `BreadcrumpIndexer`
+
+### Subscribe to an indexer event
+
+For this we need a new subscriber. If you are not familiar with a subscriber, have a look at our [Listening to events](../../plugin-fundamentals/listening-to-events.md) guide. For this example, we just write a new entry to the `log_entry` database table, indicating that a customer was updated.
+
+{% code title="<plugin root>/src/Service/Subscriber.php" %}
+```php
+<?php declare(strict_types=1);
+
+namespace Swag\BasicExample\Service;
+
+use Doctrine\DBAL\Connection;
+use Shopware\Core\Checkout\Customer\Event\CustomerIndexerEvent;
+use Shopware\Core\Framework\Uuid\Uuid;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+class Subscriber implements EventSubscriberInterface
+{
+    /**
+     * @var Connection
+     */
+    private Connection $connection;
+
+    public function __construct(Connection $connection)
+    {
+        $this->connection = $connection;
+    }
+
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            CustomerIndexerEvent::class => 'onCustomerIndexerHandle'
+        ];
+    }
+
+    public function onCustomerIndexerHandle(CustomerIndexerEvent $customerIndexerEvent)
+    {
+        foreach ($customerIndexerEvent->getIds() as $id) {
+            $this->writeLog($id);
+        }
+    }
+
+    private function writeLog($customerId)
+    {
+        $this->connection->executeStatement('INSERT INTO `log_entry` (`id`, `message`, `level`, `channel`, `created_at`) VALUES (:id, :message, :level, :channel, now())', [
+            'id' => Uuid::randomBytes(),
+            'message' => 'Updated customer with id: ' . $customerId,
+            'level' => 1,
+            'channel' => 'debug'
+        ]);
+    }
+}
+```
+The service definition for the subscriber would look like this.
+```xml
+<?xml version="1.0" ?>
+<container xmlns="http://symfony.com/schema/dic/services"
+           xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+           xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
+
+    <services>
+        <service id="Swag\BasicExample\Service\Subscriber" >
+            <argument type="service" id="Doctrine\DBAL\Connection" />
+            <tag name="kernel.event_subscriber" />
+        </service>
+    </services>
+</container>
+```
+
+It is recommended to work directly with the `Connection` since the event is dispatched in the context of an indexer. If we would use the Data Abstraction Layer \(DAL\) for writing changes to the database, the indexer could be triggered again, because it listens for `EntityWrittenContainerEvent` events. This could lead to an infinite loop. Using the `Connection` directly prevents the DAL from dispatching entity written events.
```

Comment:
```suggestion
It is recommended to work directly with the `Connection` since the event is dispatched in the context of an indexer. If we would use the Data Abstraction Layer \(DAL\) for writing changes to the database, the indexer would be triggered again, because it listens for `EntityWrittenContainerEvent` events. This would lead to an infinite loop. Using the `Connection` directly prevents the DAL from dispatching entity written events.
```
Also link to ### Use DAL functionalities in the indexer